sonar-plugin-api/src/main/java/org/sonar/api/server/profile/BuiltInQualityProfilesDefinition.java

```diff
@@ -0,0 +1,419 @@
+        ruleBuilder.put(RuleKey.of(newActiveRule.repoKey, newActiveRule.ruleKey), new BuiltInActiveRule(newActiveRule));
+      }
+      this.activeRulesByKey = unmodifiableMap(ruleBuilder);
+    }
+
+    @Override
+    public String language() {
+      return language;
+    }
+
+    @Override
+    public String name() {
+      return name;
+    }
+
+    @Override
+    public boolean isDefault() {
+      return isDefault;
+    }
+
+    @Override
+    @CheckForNull
+    public BuiltInActiveRule rule(RuleKey ruleKey) {
+      return activeRulesByKey.get(ruleKey);
+    }
+
+    @Override
+    public List<BuiltInActiveRule> rules() {
+      return unmodifiableList(new ArrayList<>(activeRulesByKey.values()));
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+      BuiltInQualityProfileImpl that = (BuiltInQualityProfileImpl) o;
+      return language.equals(that.language) && name.equals(that.name);
+    }
+
+    @Override
+    public int hashCode() {
+      int result = name.hashCode();
+      result = 31 * result + language.hashCode();
+      return result;
+    }
+
+    @Override
+    public String toString() {
+      StringBuilder sb = new StringBuilder("BuiltInQualityProfile{");
+      sb.append("name='").append(name).append('\'');
+      sb.append(", language='").append(language).append('\'');
+      sb.append(", default='").append(isDefault).append('\'');
+      sb.append('}');
+      return sb.toString();
+    }
+  }
+
+  class NewBuiltInActiveRule {
+    private final String repoKey;
+    private final String ruleKey;
+    private String overridenSeverity = null;
+    private final Map<String, NewOverridenParam> paramsByKey = new HashMap<>();
+
+    private NewBuiltInActiveRule(String repoKey, String ruleKey) {
+      this.repoKey = repoKey;
+      this.ruleKey = ruleKey;
+    }
+
+    public String repoKey() {
+      return this.repoKey;
+    }
+
+    public String ruleKey() {
+      return this.ruleKey;
+    }
+
+    public NewBuiltInActiveRule overrideSeverity(String s) {
+      checkArgument(Severity.ALL.contains(s), "Severity of rule %s is not correct: %s", RuleKey.of(repoKey, ruleKey), s);
+      this.overridenSeverity = s;
+      return this;
+    }
+
+    /**
+     * Create a parameter with given unique key. Max length of key is 128 characters.
+     */
+    public NewOverridenParam overrideParam(String paramKey, @Nullable String value) {
+      checkArgument(!paramsByKey.containsKey(paramKey), "The parameter '%s' was already overriden on the built in active rule %s", paramKey, this);
+      NewOverridenParam param = new NewOverridenParam(paramKey).setOverridenValue(value);
+      paramsByKey.put(paramKey, param);
+      return param;
+    }
+
+    @CheckForNull
+    public NewOverridenParam overridenParam(String paramKey) {
```

Comment:
When writing an implementation I mixed this method with `overrideParam(String,String)` which returns the same result type. To avoid ambiguity I suggest to rename this method to `getOverriddenParam()`.
To be consistent `overridenParams()` should be renamed too and `getOverriddenSeverity()` should be added.